### PR TITLE
Added draft on bert explanation

### PR DESCRIPTION
**Pull request by dalmeraz**
Still missing draft on experiments

### CODE REVIEW

**Comment by dalmeraz** [comment]
> Still missing draft on experiments

Done

**Comment by dalmeraz** [comment]
> Still missing draft on experiments

Done

**Review by joshpapermaster — APPROVED** [review]
Realized my review was never submitted, sorry about that

**Comment by joshpapermaster** [comment]
Had them stuck as "pending" since Thursday. Most of my comments were minor anyway though, the summary was really good.

**Comment by zhaoyue-zephyrus** [comment]
@joshpapermaster thanks for reviewing! @dalmeraz please consider revising a bit if you have time and let me know whenyou are ready for merging.

File: _summaries/devlin2018bert_1.md [hunk 1]
```diff
@@ -13,7 +13,68 @@ TODO: Summarize the paper:
 * How well does the paper perform?
 * What interesting variants are explored?
 
+
+### BERT: Bidirectional Encoder Representation from Transformers
+BERT is a simple and empirically powerful model that achieves state of the art results in a variety of both token and sentence level tasks. Its simplicity comes from only minimal changes needed to be done to use the model is separate tasks and its strength comes from its ability to look at text bidirectionally and pre-training. BERT is pre-trained and is then fine-tuned to a variety of tasks.
```
[the remark this answers is not in the document]
Comment (dalmeraz) [reply _summaries/devlin2018bert_1.md]: Done!

File: _summaries/devlin2018bert_1.md [hunk 2]
```diff
@@ -13,7 +13,68 @@ TODO: Summarize the paper:
...
+
+The fourth task was Situations With Adversarial Generations (SWAG) in which the task is that given 1 sentence, pick which of 4 sentences is a more likely continuation. This is fed to the model as 4 separate inputs and best result is given as the answer.
+
+<p align="center"> <img src="devlin2018bert_1_f.png" height="200"/> </p>
+
+The next experiments studied the effect of pre-training tasks on different metrics. LTR refers to using left to right LM as opposed to MLM.
+
+<p align="center"> <img src="devlin2018bert_1_g.png" height="200"/> </p>
+
+Finally, the final experiments looks at the effects of different model  sizes on results. One key point to note is generally, the sizes picked for BERT are considerably larger than all other transformers trained previously.
```
[the remark this answers is not in the document]
Comment (dalmeraz) [reply _summaries/devlin2018bert_1.md]: Done!

File: _summaries/devlin2018bert_1.md [hunk 3]
```diff
@@ -13,7 +13,68 @@ TODO: Summarize the paper:
...
+
+<p align="center"> <img src="devlin2018bert_1_i.png" height="200"/> </p>
+
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+* Bidirecional pre-training is important to generating good models
```
[the remark this answers is not in the document]
Comment (dalmeraz) [reply _summaries/devlin2018bert_1.md:70]: Done!

File: _summaries/devlin2018bert_1.md [hunk 4]
```diff
@@ -13,7 +13,68 @@ TODO: Summarize the paper:
...
+
+
+### Pre-training
+BERT is pre-trained through the use of unlabeled data in 2 separate tasks: Masked Language Modeling (MLM) and Next Sentence Prediction (NSP).
+
+The MLM pretraining task is what allows the model to be bidrectional. Typically, language modeling is done either left to right or right to left since doing bidirectional would allow other parts of the sequence to be seen. However BERT solves this by masking certain tokens and making those the tokens to be predicted. 15% of sequence tokens are masked and the model predicts the masked tokens. Key issue with this is that when fine-tuning the model does not recieve '[MASK]' tokens as input, so to mitigates this, 80% of the time the actual '[mask]' token is used, 10% of the a random token, and 10% the original.
+
+The NSP task was motivated by tasks such as question answering and natural language inference where it is crucial for the model to understand the relationship between two sentences. NSP is a binary prediction task in which two sentences are given and the model predicts if the second sentence follows the first.
+
+### Fine-tuning
```
[the remark this answers is not in the document]
Comment (dalmeraz) [reply _summaries/devlin2018bert_1.md:39]: Done!

File: _summaries/devlin2018bert_1.md [hunk 5]
```diff
@@ -13,7 +13,68 @@ TODO: Summarize the paper:
...
+
+<p align="center"> <img src="devlin2018bert_1_i.png" height="200"/> </p>
+
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+* Bidirecional pre-training is important to generating good models
+* Generalized pre-trained representations reduce dependencies on complex models for specific tasks
+* Above mentioned points allow for SOTA results on 11 NLP tasks
```
[the remark this answers is not in the document]
Comment (dalmeraz) [reply _summaries/devlin2018bert_1.md:72]: Thanks finished adding!

**Comment by dalmeraz** [comment]
> @joshpapermaster thanks for reviewing! @dalmeraz please consider revising a bit if you have time and let me know whenyou are ready for merging.

Sorry for accidentally closing!  Finished applying changes and ready for merging @zhaoyue-zephyrus, thanks @joshpapermaster!

**Review by zhaoyue-zephyrus — APPROVED** [review]
